app/models/user.rb

```diff
@@ -200,6 +200,21 @@ def can_make_challenges
      challenges. Please wait until they are completed before you create another."]
   end
 
+  def has_clearance?(permission_level)
+    case permission_level
+    when 2
+      is_moderator || is_admin
+    when 3
+      is_admin
+    when 4
+      is_developer || is_admin
+    when 5
+      is_developer || is_moderator || is_admin
+    else
+      true
+    end
+  end
```

Comment:
Hm. While I do acknowledge that this is probably dated and will probably revised in the future for a more robust, database-focused solution to role management, I do also think that some sanity with regards to the numbers vaguely being characterized as relative to a hierarchy would be suitable (minor, but it does help avoid console-admin mistakes).
To that end:
1: everybody (no check)
2: is_dev || is_mod || is_admin
3: is_dev || is_admin
4: is_mod || is_admin
5: is_admin

**Note: you will need to add a migration that converts the existing permission levels into their equivalent ones in this new hierarchy.**